collect_app/src/main/java/org/odk/collect/android/activities/FormHierarchyActivity.java

```diff
@@ -596,20 +602,24 @@ private void refreshView(boolean isGoingUp) {
                         break;
                     }
                     case FormEntryController.EVENT_REPEAT: {
-                        if (!formController.isGroupRelevant()) {
+                        boolean forPicker = shouldShowRepeatGroupPicker();
+                        // For #4570
+                        if (!formController.isGroupRelevant()
+                                && forPicker // Comment out this line to restore old logic
```

Comment:
Instead of referencing the issue and having this comment about deleting this line I think a comment that explains this behaviour might be more useful to someone looking at this code for the first time. Something like "Don't show repeat group in picker view if it's not relevant".

Reply:
The idea of commenting out the line clearly had to go, it 
was just a development check. I've tweaked your suggestion to reflect the original motivation.